- Description: zfcp: Wrong return code of fsf_req_create.
  Symptom:     Addressing exception, system hangs.
  Problem:     If QDIO queues are shut down fsf_req_create may return with a NULL
               pointer but a return code which indicates that this pointer is
               actually pointing to a valid structure.
  Solution:    Set return code correctly.
  Problem-ID:  10200
  Archs:       s390-31, s390-64

--- linux-2.5/drivers/s390/scsi/zfcp_fsf.c	1 Jul 2004 09:01:45 -0000	1.43.2.6
+++ linux-2.5/drivers/s390/scsi/zfcp_fsf.c	26 Jul 2004 13:23:01 -0000	1.43.2.7
@@ -29,7 +29,7 @@
  */
 
 /* this drivers version (do not edit !!! generated and updated by cvs) */
-#define ZFCP_FSF_C_REVISION "$Revision: 1.43.2.6 $"
+#define ZFCP_FSF_C_REVISION "$Revision: 1.43.2.7 $"
 
 #include "zfcp_ext.h"
 
@@ -4788,6 +4788,7 @@
 
 	if (!atomic_test_mask(ZFCP_STATUS_ADAPTER_QDIOUP, &adapter->status)) {
 		write_unlock_irqrestore(&req_queue->queue_lock, *lock_flags);
+		ret = -EIO;
 		goto failed_sbals;
 	}
 

